File: app/src/main/java/com/asafvaron/themoviedbtest/ui/mvp_snapping/SnappingContract.java
package com.asafvaron.themoviedbtest.ui.mvp_snapping;

import com.asafvaron.themoviedbtest.model.Movie;

import java.util.List;

/**
 * Created by asafvaron on 05/03/2017.
 */
interface SnappingContract {

    interface View {
        void onNowPlayingLoaded(List<Movie> movies);

        void onTopRatedLoaded(List<Movie> movies);

        void onUpcomingLoaded(List<Movie> movies);

        void onPopularLoaded(List<Movie> movies);
    }

    interface Presenter {
        void loadData(boolean forceLoad);
    }
}

File: app/src/main/java/com/asafvaron/themoviedbtest/ui/mvp_snapping/SnappingContract.kt
package com.asafvaron.themoviedbtest.ui.mvp_snapping

import com.asafvaron.themoviedbtest.model.Movie

/**
 * Created by asafvaron on 05/03/2017.
 */
internal interface SnappingContract {

    interface View {
        fun onNowPlayingLoaded(movies: List<Movie>?)

        fun onTopRatedLoaded(movies: List<Movie>?)

        fun onUpcomingLoaded(movies: List<Movie>?)

        fun onPopularLoaded(movies: List<Movie>?)
    }

    interface Presenter {
        fun loadData(forceLoad: Boolean)
    }
}
